responses/__init__.py

```diff
@@ -756,11 +781,20 @@ def remove(self, method_or_response=None, url=None):
         if isinstance(method_or_response, BaseResponse):
             response = method_or_response
         else:
+            assert url is not None
+            assert isinstance(method_or_response, str)
             response = BaseResponse(method=method_or_response, url=url)
 
         return self._registry.remove(response)
 
-    def replace(self, method_or_response=None, url=None, body="", *args, **kwargs):
+    def replace(
+        self,
+        method_or_response: Optional[Union[str, BaseResponse]] = None,
+        url: "Optional[Union[Pattern[str], str]]" = None,
```

Comment:
Should we have type aliases for this and `Optional[Union[str, BaseResponse]]`? They are both used a few times.

Reply:
good idea, I added them